### ISSUE TITLE
[Bug]: I can't set module when i using API

### ISSUE THREAD

**Issue by liuyi0501** [Mikubill/sd-webui-controlnet#974]
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits of both this extension and the webui

### What happened?

I can't set module when i using API /sdapi/v1/txt2img. It will show:                                                                 Loading preprocessor: none   
However, it will work when i using webui.

### Steps to reproduce the problem

1. start sd webui
2. post /sdapi/v1/txt2img

### What should have happened?

i don't know

### Commit where the problem happens

webui: /sdapi/v1/txt2img
controlnet: 

### What browsers do you use to access the UI ?

Google Chrome

### Command Line Arguments

```Shell
python launch.py --port 8010 --xformers --api
```

### Console logs

```Shell
sstart_unsafe.sh                                                                            Python 3.10.9 (main, Mar  1 2023, 18:23:06) [GCC 11.2.0]
Commit hash: 22bcc7be428c94e9408f589966c2040187245d81
Installing requirements for Web UI
If submitting an issue on github, please provide the full startup log for debugging purposes.                                                                                       
Initializing Dreambooth
Dreambooth revision: 926ae204ef5de17efca2059c334b6098492a0641
Successfully installed accelerate-0.18.0 fastapi-0.94.1 gitpython-3.1.31 google-auth-oauthlib-0.4.6 requests-2.28.2 transformers-4.26.1                                             
Does your project take forever to startup?
Repetitive dependency installation may be the reason.
Automatic1111's base project sets strict requirements on outdated dependencies.
If an extension is using a newer version, the dependency is uninstalled and reinstalled twice every startup.                                                                        
# Your version of xformers is 0.0.16rc425.
# xformers >= 0.0.17.dev is required to be available on the Dreambooth tab.
# Torch 1 wheels of xformers >= 0.0.17.dev are no longer available on PyPI,
# but you can manually download them by going to:
https://github.com/facebookresearch/xformers/actions
# Click on the most recent action tagged with a release (middle column).
# Select a download based on your environment.
# Unzip your download
# Activate your venv and install the wheel: (from A1111 project root)
cd venv/Scripts
activate
pip install {REPLACE WITH PATH TO YOUR UNZIPPED .whl file}
# Then restart your project.
[!] xformers version 0.0.16rc425 installed.
[+] torch version 1.13.1+cu117 installed.
[+] torchvision version 0.14.1+cu117 installed.
[+] accelerate version 0.18.0 installed.
[+] diffusers version 0.14.0 installed.
[+] transformers version 4.26.1 installed.
[+] bitsandbytes version 0.35.4 installed.


Launching Web UI with arguments: --port 8010 --xformers --api
2023-04-22 15:52:20.112896: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.                                                                                    To enable the following instructions: AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.                                                        2023-04-22 15:52:21.811953: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT                                                         Loading weights [fbcf965a62] from /home/zyt/stable-diffusion-webui/models/Stable-diffusion/anything-v4.5.ckpt                                                                       Creating model from config: /home/zyt/stable-diffusion-webui/configs/v1-inference.yaml
LatentDiffusion: Running in eps-prediction mode
DiffusionWrapper has 859.52 M params.
Loading VAE weights specified in settings: /home/zyt/stable-diffusion-webui/models/VAE/unknown.vae.pt                                                                               Applying xformers cross attention optimization.
Textual inversion embeddings loaded(1): bdap
Model loaded in 28.1s (load weights from disk: 16.0s, create model: 0.7s, apply weights to model: 1.1s, apply half(): 0.6s, load VAE: 6.8s, move model to device: 2.8s).            Running on local URL:  http://127.0.0.1:8010

To create a public link, set `share=True` in `launch()`.
Startup time: 43.7s (import torch: 6.7s, import gradio: 1.2s, import ldm: 1.9s, other imports: 1.3s, setup codeformer: 0.1s, load scripts: 2.5s, load SD checkpoint: 28.4s, create ui: 1.0s, gradio launch: 0.1s, scripts app_started_callback: 0.5s).                        Loading model: control_v11p_sd15_openpose [cab727d4]
Loaded state_dict from [/home/zyt/stable-diffusion-webui/extensions/sd-webui-controlnet/models/control_v11p_sd15_openpose.pth]
Loading config: /home/zyt/stable-diffusion-webui/extensions/sd-webui-controlnet/models/control_v11p_sd15_openpose.yaml
ControlNet model control_v11p_sd15_openpose [cab727d4] loaded.
using mask as input
Loading preprocessor: none
 82%|██████████████████████████████████████████|█▌         | 23/28 [00:15<00:02,  2.43it/s 86%|██████████████████████████████████████████ ███▍       | 24/28 [00:15<00:01,  2.43it/s 89%|██████████████████████████████████████████ █████▎     | 25/28 [00:16<00:01,  2.42it/s]otal progress:  89%|█████████████████████████████████    | 25/28 [00:09<00:01,  2.42it/s] 93%|██████████████████████████████████████████ ███████▏   | 26/28 [00:16<00:00,  2.42it/s]  | 26/28 [00:10<00:00,  2.42it/s]
 96%|██████████████████████████████████████████ █████████  | 27/28 [00:17<00:00,  2.43it/s]▋ | 27/28 [00:10<00:00,  2.43it/s]
100%|██████████████████████████████████████████ ███████████| 28/28 [00:17<00:00,  2.42it/s100%|██████████████████████████████████████████                                           ███████████| 28/28 [00:17<00:00,  1.59it/s]
```

### Additional information

The json i post to API and it respond part:
{"controlnet"{"args":[{"input_image":"","mask":"";module":"openpose","model":"control_v11p_sd15_openpose [cab727d4]","weight":1,"resize_mode":"Crop and Resize","lowvram":false,"processor_res":64,"threshold_a":64,"threshold_b":64,"guidance":1,"guidance_start":0,"guidance_end":1,"guessmode":false,"pixel_perfect":false}]}}}

**Comment by ljleb** [comment]
This line is responsible for this error:

https://github.com/Mikubill/sd-webui-controlnet/blob/68e4acf7c9ae899fd073df9cc3712cf0b7cb249d/scripts/controlnet.py#L868

Is there a reason we bypass the preprocessor from users?

**Comment by liuyi0501** [comment]
Yes, it worked. I made a copy of the "input_image" data into the "mask" and it stopped working
